pkg/controller/storagecluster/volumesnapshotterclasses.go

```diff
@@ -0,0 +1,107 @@
+	retSC := &snapapi.VolumeSnapshotClass{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: generateNameForSnapshotClass(instance, snapShotterType),
+		},
+		Driver: generateNameForSnapshotClassDriver(instance, snapShotterType),
+		Parameters: map[string]string{
+			"clusterID":                instance.Namespace,
+			snapshotterSecretName:      generateNameForSnapshotClassSecret(snapShotterType),
+			snapshotterSecretNamespace: instance.Namespace,
+		},
+		DeletionPolicy: snapapi.VolumeSnapshotContentDelete,
+	}
+	return retSC
+}
+
+func newSnapshotClasses(instance *ocsv1.StorageCluster) []*snapapi.VolumeSnapshotClass {
+	scs := []*snapapi.VolumeSnapshotClass{
+		newVolumeSnapshotClass(instance, rbdSnapshotter),
+		newVolumeSnapshotClass(instance, cephfsSnapshotter),
+	}
+	return scs
+}
+
+func (r *ReconcileStorageCluster) createSnapshotClasses(vscs []*snapapi.VolumeSnapshotClass, reqLogger logr.Logger) error {
+	for _, vsc := range vscs {
+		existing := &snapapi.VolumeSnapshotClass{}
+		err := r.client.Get(context.TODO(), types.NamespacedName{Name: vsc.Name, Namespace: vsc.Namespace}, existing)
+		if err != nil {
+			if errors.IsNotFound(err) {
+				// Since the SnapshotClass is not found, we will create a new one
+				reqLogger.Info(fmt.Sprintf("Creating SnapshotClass: %s", vsc.Name))
+				if err = r.client.Create(context.TODO(), vsc); err == nil {
+					// no error then continue to check the next snapshotter
+					continue
+				}
+			} // on any error, including (above) resource creation error, will be dealt below
+		}
+		// if there is *still* an error, we should report it
+		if err != nil {
+			reqLogger.Error(err,
+				fmt.Sprintf("Unable to Get/Create SnapshotClass: %s", vsc.Name))
+			return err
+		}
+		if existing.DeletionTimestamp != nil {
+			return fmt.Errorf("failed to restore snapshotclass  %s because it is marked for deletion", existing.Name)
+		}
+		// if there is a mis-match in the parameters of existing vs created resources,
+		if !reflect.DeepEqual(vsc.Parameters, existing.Parameters) {
+			// we have to update the existing SnapshotClass
+			reqLogger.Info(fmt.Sprintf("SnapshotClass %s needs to be updated", existing.Name))
+			existing.ObjectMeta.OwnerReferences = vsc.ObjectMeta.OwnerReferences
+			vsc.ObjectMeta = existing.ObjectMeta
+			if err := r.client.Update(context.TODO(), vsc); err != nil {
+				reqLogger.Error(err, fmt.Sprintf("SnapshotClass %s updation failed", existing.Name))
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+// ensureSnapshotClasses functions ensures that snpashotter classes are created
+func (r *ReconcileStorageCluster) ensureSnapshotClasses(instance *ocsv1.StorageCluster, reqLogger logr.Logger) error {
+	if instance.Status.SnapshotClassesCreated {
+		return nil
+	}
+	var err error
+	scs := newSnapshotClasses(instance)
+	if err = r.createSnapshotClasses(scs, reqLogger); err == nil {
+		instance.Status.SnapshotClassesCreated = true
```

Comment:
you need to run `r.Status.Update()` to write changes to server.

Reply:
@umangachapagain , we don't require a `r.Status.Update()` here as it will be done in the `Reconcile` method after each ensure function is called. As far as I could see, this is a common pattern in all the ensure functions.